R/corPFSOS.R

```diff
@@ -0,0 +1,123 @@
+# E(PFS) helpers:
+survPFS <- function(transition, t) {
+  UseMethod("survPFS")
+}
+
+survPFS.ExponentialTransition <- function(transition, t) {
+  ExpSurvPFS(t = t, h01 = transition$hazards$h01, h02 = transition$hazards$h02)
+}
+
+survPFS.WeibullTransition <- function(transition, t) {
+  WeibSurvPFS(
+    t = t, h01 = transition$hazards$h01, h02 = transition$hazards$h02,
+    p01 = transition$weibull_rates$p01, p02 = transition$weibull_rates$p02
+  )
+}
+
+survPFS.PWCTransition <- function(transition, t) {
+  PWCsurvPFS(t,
+    h01 = transition$hazards$h01, h02 = transition$hazards$h02,
+    pw01 = transition$intervals$pw01, pw02 = transition$intervals$pw02
+  )
+}
+
+# E(OS) helpers:
+survOS <- function(transition, t) {
+  UseMethod("survOS")
+}
+
+survOS.ExponentialTransition <- function(transition, t) {
+  ExpSurvOS(
+    t = t, h01 = transition$hazards$h01, h02 = transition$hazards$h02,
+    h12 = transition$hazards$h12
+  )
+}
+
+survOS.WeibullTransition <- function(transition, t) {
+  WeibSurvOS(
+    t = t, h01 = transition$hazards$h01, h02 = transition$hazards$h02,
+    h1 = transition$hazards$h1, p01 = transition$weibull_rates$p01,
+    p02 = transition$weibull_rates$p02, p12 = transition$weibull_rates$p12
+  )
+}
+
+survOS.PWCTransition <- function(transition, t) {
+  PWCsurvOS(
+    t = t, h01 = transition$hazards$h01, h02 = transition$hazards$h02,
+    h12 = transition$hazards$h12, pw01 = transition$intervals$pw01,
+    pw02 = transition$intervals$pw02, pw12 = transition$intervals$pw12
+  )
+}
+
+# Var(PFS)/Var(OS) helpers:
+expvalPFSInteg <- function(transition, t) {
+  t * survPFS(transition, t)
+}
+
+expvalOSInteg <- function(transition, t) {
+  t * survOS(transition, t)
+}
+
+# E(PFS*OS) helpers:
+
+#------- P_11 (see Meller et al, 4274)
```

Comment:
this would be important to have a methods details vignette to provide the "link" between original paper and this package's implementation. See e.g. https://github.com/openpharma/mmrm/tree/main/vignettes for inspiration on the format/flavor of these. can be done in next issue/PR